### PR TITLE
Community feedback, move to draft-iab

### PR DESCRIPTION
**Pull request by britram**

### CODE REVIEW

File: draft-iab-wire-image.md [hunk 1]
```diff
@@ -61,53 +61,69 @@ discuss in this document. It is the wire image, not the protocol's
...
-
-This document presents the wire image abstraction with the hope that it can
-shed some light on these discussions.
+The increasing deployment of transport-layer security {{?RFC8226}} to protect
+application-layer headers and payload, as well as the definition and deployment
+of QUIC {{?I-D.ietf-quic-transport}}, a transport protocol which encrypts most
+of its own control information, bring new relevance to this question. QUIC is,
+in effect, the first IETF-defined transport protocol to take care of the
+minimization of its own wire image, for reasons of ossification prevention as
+well as reducing information radiation off the protocol for end-to-end privacy.
```
Comment (tfpauly) [inline draft-iab-wire-image.md]: This last sentence seems a bit wordy.

Perhaps:
`QUIC is, in effect, the first IETF-defined transport protocol to take care of the minimization of its own wire image, to prevent ossification and improve end-to-end privacy by reducing information radiation.`

File: draft-iab-wire-image.md [hunk 2]
```diff
@@ -61,53 +61,69 @@ discuss in this document. It is the wire image, not the protocol's
...
+this draft treats the wire image as a pure abstraction, with the hope that it
+can shed some light on these discussions.
 
 # Definition
 
-More formally, the wire image of a protocol consists of the sequence of
-messages sent by each participant in the protocol, each expressed as a
-sequence of bits with an associated arbitrary-precision time at which it was
-sent.
+More formally, the wire image of a protocol stack observed at a given point
```
Comment (tfpauly) [inline draft-iab-wire-image.md]: `Wire image of a protocol stack`? Or the `wire images of a set of protocols`?

File: draft-iab-wire-image.md [hunk 3]
```diff
@@ -61,53 +61,69 @@ discuss in this document. It is the wire image, not the protocol's
...
+in the network consists of the sequence of packets sent by each participant in
+the communication, each expressed as a sequence of bits with an associated
```
Comment (tfpauly) [inline draft-iab-wire-image.md]: Nit: I think "with *the* associated... time at which it", not "with *an* associated", flows better for discussing a specific "each" packet.

File: draft-iab-wire-image.md [hunk 4]
```diff
@@ -61,53 +61,69 @@ discuss in this document. It is the wire image, not the protocol's
...
+the path(s) between the endpoints using the protocols. A protocol with a wire
 image that is largely writable operating over a path with devices that
 understand the semantics of the protocol's wire image can modify it, in order
 to induce behaviors at the protocol's participants. This is the case with TCP
 in the current Internet.
 
+For a given packet observed at a given point in the network, the wire image
+contains information from the entire stack of protocols in use at that
+observation point. Confidentiality and integrity protection may be added at
+multiple layers in the stack. However, PDUs at the transport layer and above are
```
Comment (tfpauly) [inline draft-iab-wire-image.md]: Perhaps define PDU here, since it isn't used previously?

**Review by tfpauly — COMMENTED** [review]
Looks good in general!

File: draft-iab-wire-image.md (hunk 2 above)
Comment (britram) [reply draft-iab-wire-image.md]: I'm still trying out this language; it's to address Stephen's point that you can't consider any one protocol in isolation with respect to what it exposes, you have to consider all of the protocols above and below it... I'm leaning toward "stack" here but I might need to be a little more wordy about what exactly I mean -- it's not the software on the endhost, it's the... erm, as you say, set of protocols in use at a given point.

Will continue considering.

File: draft-iab-wire-image.md (hunk 4 above)
Comment (britram) [reply draft-iab-wire-image.md]: Or not use it at all; thanks for flagging...

File: draft-iab-wire-image.md (hunk 3 above)
Comment (britram) [reply draft-iab-wire-image.md]: yep thx

**Comment by britram** [comment]
thanks for the review @tfpauly! Not *quite* yet done incorporating community feedback from the list, once I am (and I've addressed your points here) can i ping you to have another look?

File: draft-iab-wire-image.md (hunk 2 above)
Comment (britram) [reply draft-iab-wire-image.md]: ah actually, yes, it's *set* of protocols because of associated interactions (control and data, DNS, etc.)

File: draft-iab-wire-image.md (hunk 1 above)
Comment (britram) [reply draft-iab-wire-image.md]: yep done, thx.
